File: src/main/java/com/my/admin/app/commerce/model/document/ProductType.java
package com.my.admin.app.commerce.model.document;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;

@Document(collection = "product_types")
@Data
public class ProductType implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    private Long id;

    private String name;

    private String description;

    private String icon;
}

File: src/main/kotlin/com/bk/reactive/app/commerce/admin/my/model/document/ProductType.kt
package com.bk.reactive.app.commerce.admin.my.model.document

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "product_types")
data class ProductType(
    @Id
    val id: String,
    val name: String,
    val description: String,
    val icon: String
)